rslearn/config/dataset.py

```diff
@@ -332,6 +392,50 @@ class DataSourceConfig(BaseModel):
         description="Whether to ingest this layer (default True). If False, it will be directly materialized without ingestion.",
     )
 
+    @model_validator(mode="before")
+    @classmethod
+    def convert_from_legacy(cls, d: dict[str, Any]) -> dict[str, Any]:
```

Comment:
We may want to add some tests around this behavior, suggest putting them into a separate test class so they are isolating and can be easily deleted later once we remove the legacy backward compatibility code